### ISSUE TITLE
flask_session can't set SameSite 

### ISSUE THREAD

**Issue by shaMiaoMiao** [fengsp/flask-session#124]
### With the chrome's version update, the same-site value change to Strict default, if we use iframe to add no-same-site web, will find session_id can't write cookie problem.But unfortunately, flask_session seems can't set same-site, because in
- site-packages/flask_session/sessions.py 
- line 171
```python
        response.set_cookie(app.session_cookie_name, session_id,
                            expires=expires, httponly=httponly,
                            domain=domain, path=path, secure=secure)
```
not set same-site value method what it should give
so we have to rewrite Session and save_session method to fix it like this
```python
class BaseRedisSessionInterface(RedisSessionInterface):
    """rewrite RedisSessionInterface save_session method"""

    @staticmethod
    def total_seconds(td):
        """
            get seconds
        @param td:
        @return:
        """
        return td.days * 60 * 60 * 24 + td.seconds

    def save_session(self, app, session, response):
        """
            save session add samesite
        @param app:
        @param session:
        @param response:
        """
        domain = self.get_cookie_domain(app)
        path = self.get_cookie_path(app)
        if not session:
            if session.modified:
                self.redis.delete(self.key_prefix + session.sid)
                response.delete_cookie(app.session_cookie_name,
                                       domain=domain, path=path)
            return

        # Modification case.  There are upsides and downsides to
        # emitting a set-cookie header each request.  The behavior
        # is controlled by the :meth:`should_set_cookie` method
        # which performs a quick check to figure out if the cookie
        # should be set or not.  This is controlled by the
        # SESSION_REFRESH_EACH_REQUEST config flag as well as
        # the permanent flag on the session itself.
        # if not self.should_set_cookie(app, session):
        #    return

        httponly = self.get_cookie_httponly(app)
        secure = self.get_cookie_secure(app)
        expires = self.get_expiration_time(app, session)
        val = self.serializer.dumps(dict(session))
        self.redis.setex(name=self.key_prefix + session.sid, value=val,
                         time=self.total_seconds(
                             app.permanent_session_lifetime))
        if self.use_signer:
            session_id = self._get_signer(app).sign(want_bytes(session.sid))
        else:
            session_id = session.sid
        response.set_cookie(app.session_cookie_name, session_id,
                            expires=expires, httponly=httponly,
                            domain=domain, path=path,
                            secure=secure, samesite='None',
                            )
```
and use a new class such as NewSession replace Session.
```python
class NewSession(Session):
    """rewrite Session module"""
    def _get_interface(self, app):
        config = app.config.copy()
        config.setdefault('SESSION_TYPE', 'null')
        config.setdefault('SESSION_PERMANENT', True)
        config.setdefault('SESSION_USE_SIGNER', False)
        config.setdefault('SESSION_KEY_PREFIX', 'session:')
        config.setdefault('SESSION_REDIS', None)
        config.setdefault('SESSION_MEMCACHED', None)
        config.setdefault('SESSION_FILE_DIR',
                          os.path.join(os.getcwd(), 'flask_session'))
        config.setdefault('SESSION_FILE_THRESHOLD', 500)
        config.setdefault('SESSION_FILE_MODE', 384)
        config.setdefault('SESSION_MONGODB', None)
        config.setdefault('SESSION_MONGODB_DB', 'flask_session')
        config.setdefault('SESSION_MONGODB_COLLECT', 'sessions')
        config.setdefault('SESSION_SQLALCHEMY', None)
        config.setdefault('SESSION_SQLALCHEMY_TABLE', 'sessions')
        session_interface = BaseRedisSessionInterface(
            config['SESSION_REDIS'], config['SESSION_KEY_PREFIX'],
            config['SESSION_USE_SIGNER'], config['SESSION_PERMANENT'])
        return session_interface
```
I fix my problem through it.

**Comment by marcus-oscarsson** [comment]
:+1:

**Comment by maxmasetti** [comment]
same problem, please fix

**Comment by aneuhold** [comment]
Same issue here. It looks like there is an active PR that would fix it here: #116

**Comment by fengsp** [comment]
Fixed in #116
